tinytuya/core.py

```diff
@@ -885,19 +929,23 @@ def _get_socket(self, renew):
                         "socket unable to connect (timeout) - retry %d/%d",
                         retries, self.socketRetryLimit
                     )
+                    err = ERR_OFFLINE

```

Comment:
I don't know how I missed this. This stepped on the exception local variable `err` which it won't allow.  Regardless, we shouldn't have had `err` for the exception anyway since it is used within the function already.  Renaming to `e`.

See #402 - Thanks @simon-kuzin